pinot-core/src/main/java/org/apache/pinot/core/common/Operator.java

```diff
@@ -41,6 +41,8 @@
    */
   T nextBlock();
 
+  void close();
+
```

Comment:
let's dont add the close method in the main interface. as the v1 engine operator API doesn't actually hold on to any native closable resources (they are held by either the executor or the DataTable reducer)

Instead, I suggest 2 alternatives:
1. we create a `V2Operator` or `IntermediateStageOperator` API and add the `close()` method there. 
2. make the closable V2 operators (mailbox send, mailbox receive, etc) implement Closeable 
```
public class MailboxSendOperator extends BaseOperator<TransferableBlock> implements Closable {
...
```

these 2 alternative ways: (1) it provides clean separation of concern; (2) doesn't effectively refactored hundred lines of code; (3) doesn't require us to move the non-native closable resources in V1 engine over to the proper close() method --> thus avoid confusion in future code maintenance. 

personally I prefer the 2nd option.